### PR TITLE
Add Bors guide

### PR DESCRIPTION
**Pull request by curquiza**
of course, the next step is to add a link to this guide in our CONTRIBUTING.md 😉

### CODE REVIEW

File: guides/bors.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+But most of all rebasing is the only way to guarantee that the work that is going to be merged fits exactly the current work in production. If the tests pass on an up-to-date branch, you ensure the tests will pass once merged into `master`, so your production work will not be broken.
+
+## How does Bors work?
+
+Bors is a bot that automatically applies a rebase before passing the tests and merging.<br>
+This allows us to _automatically maintain a repository of code that always passes all the tests__ (cf [this article](https://graydon2.dreamwidth.org/1597.html) by the creator of Bors)
+
+### Usage
+
+All the Bors commands have to be written as a comment of the PR.
```
Comment (bidoubiwa) [inline guides/bors.md:28]:
```suggestion
All the Bors commands have to be given through comments in the PR.
```

File: guides/bors.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+bors merge
+```
+
+![bors merge](../assets/screenshots/bors_merge.png)
+
+Bors is going to check the PR passes the tests when up-to-date with `master` and then merge the rebased work.
+
+💡 Bors is also able to manage multiple merging PRs at the same time. Each PR is added to a queue and will be checked with the last version on `master`.
+
+### More details
```
Comment (bidoubiwa) [inline guides/bors.md:57]:
```suggestion
### More Details
```

**Review by bidoubiwa — CHANGES_REQUESTED** [review]

File: guides/bors.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Bors: our Merge Bot for PRs
+
+[Bors](https://github.com/bors-ng/bors-ng) is a tool that helps us manage pull requests (PR) merging.
+
+## Motivations
+
+One of our requirements to merge a PR is the branch has to be up-to-date with the `master` branch.<br>
```
Comment (eskombro) [inline guides/bors.md:7]:
```suggestion
One of our requirements to merge a PR is that a contributor branch has to be up-to-date with the upstream `master` branch.<br>
```

File: guides/bors.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+The only proper way to keep the Git history clear is to rebase the branch of the PR.
```
Comment (eskombro) [inline guides/bors.md:8]:
```suggestion
The only proper way to keep the Git history clear is to rebase the branch originating the PR.
```

File: guides/bors.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+It became tedious to always ask the contributors to rebase their branch when their branch was not up-to-date with `master` because:
+- It forces everyone to wait again before the merge
+- It is not always easy to successfully apply a rebase without any issue
+
```
Comment (eskombro) [inline guides/bors.md:13]:
```suggestion
- Sometimes there are several PRs ready to be merged, which introduces complexity in coordination

```

File: guides/bors.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+### But why rebasing?
+
+The goal of a rebase is to move the commits (from the PR branch) right after the last commit on `master`.<br>
+Therefore, rebasing keeps your Git history clear and consistent with the order in which you have merged your PRs.
+
+But most of all rebasing is the only way to guarantee that the work that is going to be merged fits the current work in production. If the tests pass on an up-to-date branch, it ensures the tests will pass once merged into `master`, assuring your production branch will not break.
```
Comment (eskombro) [inline guides/bors.md:19]:
```suggestion
But most of all, rebasing is the only way to guarantee that the work that is going to be merged fits the current work in production. If the tests pass on an up-to-date branch, it ensures the tests will pass once merged into `master`, assuring your production branch will not break.
```

File: guides/bors.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+## How does Bors work?
+
+Bors is a bot that automatically applies a rebase before passing the tests and merging.<br>
```
Comment (eskombro) [inline guides/bors.md:23]:
```suggestion
Bors is a bot that automatically applies a rebase before running the tests and finally merging if they succeed.<br>
```

**Review by eskombro — CHANGES_REQUESTED** [review]
I did a few suggestions, but in general, I think it is a great guide! Thanks!!!

File: guides/bors.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
+# Bors: our Merge Bot for PRs
```
[the remark this answers is not in the document]
Comment (eskombro) [reply guides/bors.md:1]: I don't know exactly how to write it, but wouldn't go with "our bot"

File: guides/bors.md (hunk 5 above)
Comment (curquiza) [reply guides/bors.md:13]: Do you want to remove all of this paraph by your sentence?

File: guides/bors.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (curquiza) [reply guides/bors.md:1]:
```suggestion
# Our Usage of Bors
```

[the remark this answers is not in the document]
Comment (eskombro) [reply guides/bors.md:1]:
```suggestion
# Bors: a Merge Bot for PRs
```

**Review by eskombro — DISMISSED** [review]

**Pull request merged**